dove/src/cmd/prover.rs

```diff
@@ -83,18 +138,57 @@ fn is_z3_available(z3_exe: &str) -> bool {
     is_executable_available(z3_exe, &["-h"])
 }
 
+fn is_cvc4_available(cvc4_exe: &str) -> bool {
+    is_executable_available(cvc4_exe, &["-V"])
+}
+
 /// Checks if executable is available in path by running it.
 fn is_executable_available<S: AsRef<OsStr>, I: IntoIterator<Item = S>>(
     executable: &str,
     args: I,
 ) -> bool {
-    let status = std::process::Command::new(executable)
-        .args(args)
-        .stdout(Stdio::null())
-        .stderr(Stdio::null())
-        .status();
-    match status {
-        Ok(status) => status.success(),
+    let result = std::process::Command::new(executable).args(args).output();
+
+    match result {
+        Ok(result) => {
+            let status_success = result.status.success();
+            if !status_success {
+                println!("Warning: {}", String::from_utf8(result.stderr).unwrap());
+            }
+            result.status.success()
+        }
         Err(_) => false,
     }
 }
+
+/// get path to z3
+fn get_path_z3() -> Result<String, anyhow::Error> {
```

Comment:
Why `z3` is the only one that gets a special method while others just use simple constants?

Reply:
@vitvakatu He's the only one who needs a path